lib/aruba/platform.rb

```diff
@@ -182,19 +183,16 @@ def simple_table(hash)
     # rubocop:disable Metrics/MethodLength
     # rubocop:disable Metrics/CyclomaticComplexity
     def which(program, path = ENV['PATH'])
-      on_windows = false
-      on_windows = true if File::ALT_SEPARATOR
-
       program = program.to_s
```

Comment:
@dg-ratiodata If there still needs to be  ```which```  functionality, then it should be renamed to something like  ```command_valid?``` because ```which``` is a unix-specific term (command).    That is what is really happening in that method:  doing whatever is necessary to check that the program is valid so that we can send it along to ChildProcess and know it will work.  Using that name --```command_valid?```  or some variation -- means that we can do all of the checks that are necessary -- not just checking if it's on the path. We also have to ensure it's executable, for example.  This allows platform-specific checks as well as platform-specific implementations of the checks.
Someone from a Windows point of view reading the code won't understand why the method is called ```which```.  There is no ```which``` on Windows; that is meaningless to a Windows person.  On Windows there's a ```where``` command to accomplish nearly the same thing. (Actually it does exactly what we want in this situation, so the implementation for Windows becomes nearly one line: ```  result = `where #{program.to_s}` ``` )  But the command shouldn't be called ```where``` either, for the same reason it shouldn't be called ```which```.  
The method name should reflect the behavior, not the platform-specific name (or implementation) for the behavior.  
So just as you've sketched out above, there could be platform-specific classes whose sole responsibility is to validate the command before it gets sent along to create the appropriate process.  I'd suggest that the name of the classes be ```[OS]CommandValidator``` or something like that.  

###### How does making a change like this fit into what you're doing now?   
 I don't think it should be made with this PR; I suspect that either you should do it as part of the re-working of Platform that you're doing now, or do this change after your work. I just want to try to keep work coordinated.  You and @mattwynne  certainly are the ones to make the call on that.